utopia-vscode-common/src/mailbox.ts

```diff
@@ -113,10 +118,25 @@ async function waitForPathToExist(path: string, maxWaitTime: number = 5000): Pro
   }
 }
 
+async function checkAndResetIfMailboxCleared(mailbox: Mailbox): Promise<void> {
+  const mailboxClearedAtTimestamp = await getMailboxClearedAtTimestamp(inbox)
+  if (mailboxClearedAtTimestamp > mailboxLastClearedTimestamp) {
+    // The mailbox was cleared since we last polled it, meaning our last consumed message
+    // count is now invalid, and we need to start consuming messages from the beginning again.
+    lastConsumedMessage = -1
```

Comment:
I think I'd feel better if this was set to `null` or some other sentinel value like `'never'` to indicate a reset, something that actually indicates the meaning we intend.

Reply:
Good shout. I'm going to do this separately, as I need to merge this now so that I can pull it into the scenes experiment PRs.